src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
+    if (locations.length > 0 && isInitialFit) {
+      const coordinates = locations.map((loc) => [loc.position.longitude, loc.position.latitude]);
+      if (locations.length === 1) {
+        if (cameraRef.current) {
+          cameraRef.current.setCamera({
+            centerCoordinate: coordinates[0],
+            zoomLevel: 15,
+            animationDuration: 0
+          });
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      } else {
+        const [minLng, minLat] = coordinates.reduce(
+          (prev, curr) => [Math.min(prev[0], curr[0]), Math.min(prev[1], curr[1])],
+          [Infinity, Infinity]
+        );
+        const [maxLng, maxLat] = coordinates.reduce(
+          (prev, curr) => [Math.max(prev[0], curr[0]), Math.max(prev[1], curr[1])],
+          [-Infinity, -Infinity]
+        );
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0 // animationDuration in ms
+          );
+          if (clusteringEnabled) {
+            setTimeout(() => {
+              setIsLoading(false);
+            }, 1000); // Timeout in Millisekunden anpassen je nach Bedarf
+            setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+          } else {
+            setTimeout(() => {
+              setIsLoading(false);
+            }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+            setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+          }
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit, clusteringEnabled]);
+
+  useEffect(() => {
+    if (updatedRegion && cameraRef.current) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [updatedRegion.longitude, updatedRegion.latitude],
+        zoomLevel: 14,
+        animationDuration: 1000
+      });
+    }
+  }, [updatedRegion]);
+
+  const handleMapPress = (event) => {
+    if (onMapPress) {
+      const { geometry } = event;
+      const nativeEvent = {
+        coordinate: {
+          latitude: geometry.coordinates[1],
+          longitude: geometry.coordinates[0]
+        }
+      };
+      onMapPress({ nativeEvent });
+    }
+  };
+
+  const handleClusterPress = useCallback(
+    (cluster) => {
+      if (isAnimating) return; // Skip if animation is already in progress
+
+      setIsAnimating(true);
+      const [longitude, latitude] = cluster.geometry.coordinates;
+      const zoom = Math.min(zoomLevel + 2, 20); // Increase the zoom level by 2 or set to max zoom level (20)
```

Comment:
what happens when pressing a cluster? it zooms the map by 2 and stops? or does it increase until markers are visible?

Reply:
it zooms in by 2 and stops, should it zoom in until the clustered markers are visible?

Reply:
this was the last maps function, so i think so, yes. where does the "zoom in by 2" comes from? does other maps do it this way?

Reply:
"zoom in by 2" was an idea of mine to handle it and every map handles it differently. 
in the upcoming commit I've changed it to the behaviour we are used to to zoom in until the cluster & markers update/change